src/appmixer/servicenow/record/CreateRecord/component.json

```
@@ -1,53 +1,74 @@
-                "tableName": { "type": "string" }
-            },
-            "required": ["tableName"]
-        },
-        "inspector": {
-            "inputs": {
-                "tableName": {
-                    "type": "text",
-                    "label": "Table Name",
-                    "index": 1,
-                    "tooltip": "Name of the table to retrieve the record from."
-                }
-            }
-        }
-    },
     "inPorts": [
         {
             "name": "in",
-            "source": {
-                "url": "/component/appmixer/servicenow/record/CreateRecord?outPort=out",
-                "data": {
-                    "messages": {
-                        "in/sysparm_fields": "inputs/in/sysparm_fields"
-                    },
-                    "properties": {
-                        "generateInspector": true,
-                        "tableName": "properties/tableName"
-                    }
-                }
+            "schema": {
+                "type": "object",
+                "properties": {
+                    "tableName": { "type": "string" },
+                    "record": { "type": "string" },
+                    "sysparm_display_value": { "type": "boolean" },
+                    "sysparm_exclude_reference_link": { "type": "boolean" },
+                    "sysparm_fields": { "type": "string" },
+                    "sysparm_view": { "type": "string" },
+                    "sysparm_query_no_domain": { "type": "boolean" }
+                },
+                "required": [
+                    "tableName", "record"
+                ]
             },
             "inspector": {
                 "inputs": {
+                    "tableName": {
+                        "type": "text",
+                        "label": "Table name",
+                        "index": 1,
+                        "tooltip": "Name of the table from which to retrieve the record."
+                    },
+                    "record": {
+                        "type": "textarea",
+                        "label": "Record",
+                        "index": 2,
+                        "tooltip": "JSON object containing field values for the new record. Example: <code>{\"email\":\"foo@bar.baz\",\"name\":\"Foo\"}</code>."
+                    },
+                    "sysparm_display_value": {
+                        "type": "toggle",
+                        "label": "Display value",
+                        "index": 3,
+                        "tooltip": "Return field display values (true), actual values (false), or both (all) (default: false)"
+                    },
+                    "sysparm_exclude_reference_link": {
+                        "type": "toggle",
+                        "label": "Exclude reference link",
+                        "index": 4,
+                        "tooltip": "True to exclude Table API links for reference fields (default: false)"
+                    },
```

```suggestion
                    "sysparm_exclude_reference_link": {
                        "type": "toggle",
                        "label": "Exclude reference link",
                        "index": 4,
                        "tooltip": "True to exclude Table API links for reference fields",
                        "defaultValue": false
                    },
```
How about having the default value set on our side too?

---

good catch, I'll add the default value